db/pebble/iterator.go

```diff
@@ -19,12 +19,20 @@ func (i *iterator) Valid() bool {
 
 // Key : see db.Transaction.Iterator.Key
 func (i *iterator) Key() []byte {
-	return i.iter.Key()
+	key := i.iter.Key()
+	if key == nil {
+		return nil
+	}
+	return append([]byte{}, key...)
```

Comment:
Might be a micro-optimization, but what do you think about using `copy` instead of `append`?

Reply:
Done